python/hsfs/core/tfdata_engine.py

```diff
@@ -0,0 +1,468 @@
+        self._cycle_length = cycle_length
+
+        self._features = training_dataset._features
+        self._training_dataset_format = self._training_dataset.data_format
+
+        self._input_files = self._get_training_dataset_files(
+            self._training_dataset.location,
+            self._split,
+            filter_empty=True if self._training_dataset_format == "csv" else False,
+        )
+
+        if self._feature_names is None:
+            self._feature_names = [feat.name for feat in self._features]
+
+            if target_name in self._feature_names:
+                self._feature_names.remove(target_name)
+
+    def get_serialized_example_schema(self):
+        """
+        returns schema for parsing serialized example of tfrecord files
+        """
+        _, tfrecord_feature_description = self._get_tfdataset(
+            self._input_files, self._cycle_length
+        )
+        return tfrecord_feature_description
+
+    def tf_record_dataset(
+        self,
+        batch_size: Optional[int] = None,
+        num_epochs: Optional[int] = None,
+        one_hot_encode_labels: Optional[bool] = False,
+        num_classes: Optional[int] = None,
+        process: Optional[bool] = False,
+        serialized_ndarray_fname: Optional[list] = [],
+    ):
+        """
+        Reads tfrecord files and returns `ParallelMapDataset` or `PrefetchDataset` object, depending on `process` set
+        to `False` or `True`, respectively.
+
+        If `process` set to `False` returned object `ParallelMapDataset` can be further processed by user. For example
+        applying custom transformations to features, batching, caching etc. `process=True` will return `PrefetchDataset`
+        object, that contains tuple of feature vector and label, already batched and ready to input into model training.
+
+        !!! example "Example of using tf_record_dataset:"
+            ```python
+            connection = hsfs.connection()
+            fs = connection.get_feature_store();
+            td = fs.get_training_dataset("sample_model", 3)
+            td.tf_data(target_name = "id").tf_record_dataset(batch_size=1, num_epochs=1, process=True)
+            ```
+
+        # Arguments
+            batch_size: Size of batch, defaults to `None`.
+            num_epochs: Number of epochs to train, defaults to `None`.
+            one_hot_encode_labels: If set true then one hot encode labels, defaults to `False`.
+            num_classes: Iif above true then provide number of target classes, defaults to  `None`
+            process: if set true api will optimise tf data read operation, and return feature vector for model
+                with single input, defaults to `False`.
+            serialized_ndarray_fname: Names of features that contain serialised multi dimensional arrays, defaults to `[]`
+        # Returns
+           `PrefetchDataset` if `process` is set to `True`
+           `ParallelMapDataset` if `process` is set to `False`
```

Comment:
add a new line and full stops please
```suggestion

        # Returns
           `PrefetchDataset`. If `process` is set to `True`.
           `ParallelMapDataset`. If `process` is set to `False`.
```